### ISSUE TITLE
Performance Crash on 2nd Stage Training

### ISSUE THREAD

**Issue by zhangry868** [microsoft/GUI-Actor#16]
I am using a subset to reproduce the results. 1st stage warm-up training works well and the results are reasonable (Please see attached).

However, I find the 2nd stage training does not improve the performance and even damage the model performance.

After warm-up training:

```markdown
|                | mobile-text | mobile-icon | mobile-avg | desktop-text | desktop-icon | desktop-avg | web-text | web-icon | web-avg | All-text | All-icon | All-avg |
|----------------|-------------|-------------|------------|--------------|--------------|-------------|----------|----------|---------|----------|----------|---------|
| hit_top1       | 80.23       | 73.30       | 77.28      | 88.89        | 61.11        | 77.45       | 83.21    | 58.65    | 71.95   | 83.57    | 64.26    | 75.16   |
| overlap_top1   | 100.00      | 100.00      | 100.00     | 100.00       | 100.00       | 100.00      | 100.00   | 100.00   | 100.00  | 100.00   | 100.00   | 100.00  |
| hit_topk       | 94.96       | 82.20       | 89.53      | 92.78        | 69.84        | 83.33       | 91.07    | 66.24    | 79.69   | 92.90    | 72.56    | 84.04   |
| overlap_topk   | 100.00      | 100.00      | 100.00     | 100.00       | 100.00       | 100.00      | 100.00   | 100.00   | 100.00  | 100.00   | 100.00   | 100.00  |
```

After full finetuning:

```markdown
|                | mobile-text | mobile-icon | mobile-avg | desktop-text | desktop-icon | desktop-avg | web-text | web-icon | web-avg | All-text | All-icon | All-avg |
|----------------|-------------|-------------|------------|--------------|--------------|-------------|----------|----------|---------|----------|----------|---------|
| hit_top1       | 7.36        | 6.28        | 6.90       | 11.11        | 4.76         | 8.50        | 3.57     | 4.22     | 3.87    | 6.82     | 5.05     | 6.05    |
| overlap_top1   | 100.00      | 100.00      | 100.00     | 100.00       | 100.00       | 100.00      | 100.00   | 100.00   | 100.00  | 100.00   | 100.00   | 100.00  |
| hit_topk       | 17.05       | 19.37       | 18.04      | 19.44        | 8.73         | 15.03       | 12.14    | 11.39    | 11.80   | 15.74    | 13.54    | 14.78   |
| overlap_topk   | 100.00      | 100.00      | 100.00     | 100.00       | 100.00       | 100.00      | 100.00   | 100.00   | 100.00  | 100.00   | 100.00   | 100.00  |
```

**Comment by zhangry868** [comment]
```
#!/bin/bash
# model_type: qwen2vl or qwen25vl
model_type="qwen25vl"
llm_model="Qwen/Qwen2.5-VL-3B-Instruct"
output_dir="/mnt/localssd/checkpoints/${model_type}_warmup"

# === Training Command ===
torchrun --nproc_per_node=8 train.py \
  --deepspeed ./scripts/zero3.json \
  --data_path data/data_config.yaml \
  --image_folder "" \
  --model_type ${model_type} \
  --model_name_or_path ${llm_model} \
  --group_by_modality_length True \
  --bf16 True \
  --output_dir ${output_dir} \
  --num_train_epochs 1 \
  --per_device_train_batch_size 1 \
  --per_device_eval_batch_size 4 \
  --gradient_accumulation_steps 1 \
  --eval_strategy "no" \
  --save_strategy "steps" \
  --save_steps 2000 \
  --learning_rate 1e-4 \
  --weight_decay 0. \
  --warmup_ratio 0.03 \
  --lr_scheduler_type "cosine" \
  --logging_steps 10 \
  --tf32 True \
  --model_max_length 24576 \
  --gradient_checkpointing True \
  --dataloader_num_workers 8 \
  --max_pixels 5720064 \
  --unfreeze_all_parameters False \
  --unfreeze_pointer_head True \
  --unfreeze_lm_head False \
  --unfreeze_base_model False \
  --unfreeze_last_n_layers -1 \
  --unfreeze_new_tokens True \
  --unfreeze_visual False \
  --pointer_loss_weight 1.0 \
  --lm_loss_weight -1.0

```



```
#!/bin/bash
# model_type: qwen2vl or qwen25vl
model_type="qwen25vl"
llm_model="/mnt/localssd/checkpoints/${model_type}_warmup"
output_dir="/mnt/localssd/checkpoints/${model_type}_sft_test"

# === Training Command ===
torchrun --nproc_per_node=8 train.py \
  --deepspeed ./scripts/zero3.json \
  --data_path data/data_config.yaml \
  --image_folder "" \
  --model_type ${model_type} \
  --model_name_or_path ${llm_model} \
  --group_by_modality_length True \
  --bf16 True \
  --output_dir ${output_dir} \
  --num_train_epochs 1 \
  --per_device_train_batch_size 1 \
  --per_device_eval_batch_size 4 \
  --gradient_accumulation_steps 1 \
  --eval_strategy "no" \
  --save_strategy "steps" \
  --save_steps 2000 \
  --learning_rate 1e-4 \
  --weight_decay 0. \
  --warmup_ratio 0.03 \
  --lr_scheduler_type "cosine" \
  --logging_steps 10 \
  --tf32 True \
  --model_max_length 24576 \
  --gradient_checkpointing True \
  --dataloader_num_workers 8 \
  --max_pixels 5720064 \
  --unfreeze_all_parameters True \
  --unfreeze_pointer_head False \
  --unfreeze_lm_head False \
  --unfreeze_base_model False \
  --unfreeze_last_n_layers -1 \
  --unfreeze_new_tokens False \
  --unfreeze_visual False \
  --pointer_loss_weight 1.0 \
  --lm_loss_weight 1.0

```

**Comment by QianhuiWu** [comment]
@zhangry868 Many thanks for sharing the information.

The learning rate for the training after warmup should be set to a smaller value such as `5e-6`. I've just updated the scripts accordingly.
